src/math/mod.rs

```diff
@@ -124,170 +111,81 @@ where
-    type Output = Self;
-    fn mul(self, _rhs: Self) -> Self {
-        Self::new(
-            self.rotation * _rhs.rotation,
-            self.rotation * _rhs.translation + self.translation,
-        )
-    }
-}
-
-impl<'a, S: BaseFloat> Mul for &'a Isometry3<S> {
-    type Output = Isometry3<S>;
-    fn mul(self, _rhs: &'a Isometry3<S>) -> Isometry3<S> {
-        Isometry3::new(
-            self.rotation * _rhs.rotation,
-            self.rotation * _rhs.translation + self.translation,
-        )
-    }
-}
-
-impl<S: BaseFloat> Mul<Point3<S>> for Isometry3<S> {
-    type Output = Point3<S>;
-    fn mul(self, _rhs: Point3<S>) -> Point3<S> {
-        Point3::from_vec(self.rotation * _rhs.to_vec()) + self.translation
-    }
-}
-
-impl<'a, S: BaseFloat> Mul<&'a Point3<S>> for &'a Isometry3<S> {
-    type Output = Point3<S>;
-    fn mul(self, _rhs: &'a Point3<S>) -> Point3<S> {
-        Point3::from_vec(self.rotation * _rhs.to_vec()) + self.translation
-    }
-}
-
-impl<S: BaseFloat> Into<Decomposed<Vector3<S>, Quaternion<S>>> for Isometry3<S> {
-    fn into(self) -> Decomposed<Vector3<S>, Quaternion<S>> {
-        Decomposed {
-            scale: S::one(),
-            rot: self.rotation,
-            disp: self.translation,
-        }
-    }
-}
 
-impl<S: BaseFloat> From<Decomposed<Vector3<S>, Quaternion<S>>> for Isometry3<S> {
-    fn from(decomposed: Decomposed<Vector3<S>, Quaternion<S>>) -> Self {
-        Self::new(decomposed.rot, decomposed.disp)
+    fn intersects_aabb(&self, _aabb: &Aabb<S>) -> bool {
+        true
     }
 }
 
-impl<S: BaseFloat> From<nalgebra::Isometry3<S>> for Isometry3<S>
+pub fn cell_union_intersects_aabb<S>(
+    cell_union: &CellUnion,
+    aabb: &crate::geometry::Aabb<S>,
+) -> Relation
 where
-    S: nalgebra::RealField,
+    S: RealField,
+    f64: From<S>,
 {
-    fn from(isometry: nalgebra::Isometry3<S>) -> Self {
-        let r = isometry.rotation.coords;
-        let t = isometry.translation;
-        Self::new(
-            // nalgebra quaternion vec has `[ x, y, z, w ]` storage order
-            Quaternion::new(r[3], r[0], r[1], r[2]),
-            Vector3::new(t.x, t.y, t.z),
-        )
-    }
-}
-
-impl<S: BaseFloat> Isometry3<S> {
-    pub fn new(rotation: Quaternion<S>, translation: Vector3<S>) -> Self {
-        Isometry3 {
-            rotation,
-            translation,
-        }
-    }
-
-    /// Returns the identity transformation.
-    pub fn one() -> Self {
-        Isometry3 {
-            rotation: Quaternion::one(),
-            translation: Vector3::zero(),
-        }
-    }
-
-    pub fn inverse(&self) -> Self {
-        Self::new(
-            self.rotation.conjugate(),
-            -(self.rotation.conjugate() * self.translation),
-        )
+    let aabb_corner_cells = aabb
+        .compute_corners()
+        .iter()
+        .map(|p| CellID::from_point(p))
+        .collect();
+    let mut aabb_cell_union = CellUnion(aabb_corner_cells);
+    aabb_cell_union.normalize();
```

Comment:
nit: combine these two lines into one?
```suggestion
    let aabb_cell_union = CellUnion(aabb_corner_cells).normalize();
```

Reply:
The function is unfortunately written as a mutation, i.e. it returns `()`.